File: SDKs/amazonaws.com/gamelift/2015-10-01/go/pkg/models/operations/describematchmaking.go
package operations

import (
	"openapi/pkg/models/shared"
)

type DescribeMatchmakingXAmzTargetEnum string

const (
	DescribeMatchmakingXAmzTargetEnumGameLiftDescribeMatchmaking DescribeMatchmakingXAmzTargetEnum = "GameLift.DescribeMatchmaking"
)

type DescribeMatchmakingHeaders struct {
	XAmzAlgorithm     *string                           `header:"style=simple,explode=false,name=X-Amz-Algorithm"`
	XAmzContentSha256 *string                           `header:"style=simple,explode=false,name=X-Amz-Content-Sha256"`
	XAmzCredential    *string                           `header:"style=simple,explode=false,name=X-Amz-Credential"`
	XAmzDate          *string                           `header:"style=simple,explode=false,name=X-Amz-Date"`
	XAmzSecurityToken *string                           `header:"style=simple,explode=false,name=X-Amz-Security-Token"`
	XAmzSignature     *string                           `header:"style=simple,explode=false,name=X-Amz-Signature"`
	XAmzSignedHeaders *string                           `header:"style=simple,explode=false,name=X-Amz-SignedHeaders"`
	XAmzTarget        DescribeMatchmakingXAmzTargetEnum `header:"style=simple,explode=false,name=X-Amz-Target"`
}

type DescribeMatchmakingRequest struct {
	Headers DescribeMatchmakingHeaders
	Request shared.DescribeMatchmakingInput `request:"mediaType=application/json"`
}

type DescribeMatchmakingResponse struct {
	ContentType                string
	DescribeMatchmakingOutput  *shared.DescribeMatchmakingOutput
	InternalServiceException   *interface{}
	InvalidRequestException    *interface{}
	StatusCode                 int64
	UnsupportedRegionException *interface{}
}

File: SDKs/amazonaws.com/gamelift/2015-10-01/typescript/src/sdk/models/operations/describematchmaking.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";


export enum DescribeMatchmakingXAmzTargetEnum {
    GameLiftDescribeMatchmaking = "GameLift.DescribeMatchmaking"
}


export class DescribeMatchmakingHeaders extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Algorithm" })
  xAmzAlgorithm?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Content-Sha256" })
  xAmzContentSha256?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Credential" })
  xAmzCredential?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Date" })
  xAmzDate?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Security-Token" })
  xAmzSecurityToken?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Signature" })
  xAmzSignature?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-SignedHeaders" })
  xAmzSignedHeaders?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Target" })
  xAmzTarget: DescribeMatchmakingXAmzTargetEnum;
}


export class DescribeMatchmakingRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  headers: DescribeMatchmakingHeaders;

  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  request: shared.DescribeMatchmakingInput;
}


export class DescribeMatchmakingResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  describeMatchmakingOutput?: shared.DescribeMatchmakingOutput;

  @SpeakeasyMetadata()
  internalServiceException?: any;

  @SpeakeasyMetadata()
  invalidRequestException?: any;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  unsupportedRegionException?: any;
}
